libs/mgjson/Makefile

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright (C) 2021 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=mgjson
+PKG_RELEASE:=1
+
+PKG-BRANCH=main
+PKG_SOURCE_URL=https://github.com/TheRootED24/mgjson.git
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_DATE:=2024-08-17
+
+PKG_SOURCE_VERSION:=9793e7ba97f1197909dda8fa3002673e4dd7266b
+PKG_MIRROR_HASH:=85c5b3c3704e0b834f4d315b345baadd79a5e972bf688f46269d2887d4ae6acd
+
+PKG_MAINTAINER:=Scott Mercer <TheRootED24@gmail.com>
+
+PKG_LICENSE:=GPL-2.0-only
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/mgjson
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=A Stripped down OpenWrt version of Mongoose C Networking. mgjson provides only the mg_json, mg_str, and mg_iobuf, functionilty of the Mongoose Library.
+endef
+
+TARGET_CFLAGS += -std=gnu99 -I$(STAGING_DIR)/usr/include/mgjson/
+
+define Build/InstallDev
+	$(INSTALL_DIR) $(1)/usr/include/mgjson
+	$(CP) \
+		$(PKG_BUILD_DIR)/src/*.h \
+		$(1)/usr/include/mgjson/
+
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) \
+		$(PKG_BUILD_DIR)/src/libmgjson.so \
+		$(1)/usr/lib/
+	
+endef
```

Comment:
Why do you need to have this?

Reply:
As i said in my original comment, this is just a shared library, used by the second package i submitted (jsonmg) to link against, therefore the includes from mgjson are required(mgjson.h), and the libmgjson.so in required at runtime.

Reply:
I would like to give the user the option to compile static, or dynamically. given the size, static is likely the best option,  I thought there was some way i could offer this in the menuconfig ...  however i couldn't find it in the docs and again ... no answer in the forums

Reply:
i just was pointing here --> [https://openwrt.org/docs/guide-developer/frequent-pr-mistakes](url)

This would've been of great help.I see allot of mistakes could've been avoided.  My question now is. What is expected of me now, do I make the edit/s and submit a new pr, do you make the edits ? 

I agree with all proposed changes ...other than, the necessity of the inclusion of the #define Build/dev directives, which unless I am mistaken, is a valid inclusion ??

Reply:
In regards to adding more general info to the mgjson repo. Is this something you want done during review. I just do not what to add unnecessary commits during review. Please advise.